pika/heartbeat.py

```diff
@@ -165,23 +169,40 @@ def _send_heartbeat_frame(self):
             self._new_heartbeat_frame())
         self._heartbeat_frames_sent += 1
 
-    def _setup_timer(self):
-        """Use the connection objects delayed_call function which is
-        implemented by the Adapter for calling the check_heartbeats function
-        every interval seconds.
+    def _setup_timers(self):
+        """Use the connection objects _adapter_add_timeout function which is
+        implemented by the Adapter for calling our send and check functions
+        after interval seconds.
+
+        """
+        self._setup_send_timer()
+        self._setup_check_timer()
+
+    def _setup_send_timer(self): # pylint: disable=C0111
+        self._send_timer = self._connection._adapter_add_timeout(  # pylint: disable=W0212
+            self._send_interval,
+            self.send_heartbeat)
+
+    def _setup_check_timer(self): # pylint: disable=C0111
+        self._check_timer = self._connection._adapter_add_timeout(  # pylint: disable=W0212
+            self._check_interval,
+            self.check_heartbeat)
+
+    def _start_send_timer(self):
+        """If the connection still has this object set for heartbeats, add a
+        new send timer.
 
         """
-        self._timer = self._connection._adapter_add_timeout(  # pylint: disable=W0212
-            self._interval,
-            self.send_and_check)
+        if self.active:
+            self._setup_send_timer()
 
-    def _start_timer(self):
+    def _start_check_timer(self):
         """If the connection still has this object set for heartbeats, add a
-        new timer.
+        new check timer.
 
         """
         if self.active:
-            self._setup_timer()
+            self._setup_check_timer()
```

Comment:
There is unnecessary complication from the super-granular separation between _start_check_timer/_setup_check_timer and _start_send_timer/_setup_send_timer as well as _setup_timers. Let's just have _start_check_timer and _start_send_timer and no `self.active` check (as explained in the following paragraph)

And the `self.active` checks are unnecessary since `Connection` calls the heartbeat's `stop()` method when it's not needed any more, and the stop() cancels the timeouts. So, let's get rid of the redundant `HearbeatChecker.active` property as well.

These unnecessary legacy things are an unwelcome distraction for maintainers. :)